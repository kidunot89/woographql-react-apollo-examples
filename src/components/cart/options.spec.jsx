import React from 'react';
import '@testing-library/react/dont-cleanup-after-each';
import { render, cleanup } from '@testing-library/react';

import { mockProduct } from '../../testing/mocks';
import Options from './options';
import ApolloMockProvider from '../../testing/apollo-mock-provider';

describe('Options', () => {
  afterAll(() => cleanup());
  const { baseElement } = render(
    <ApolloMockProvider>
      <Options product={mockProduct} />
    </ApolloMockProvider>,
  );

  it('should render successfully', () => {
    expect(baseElement).toMatchSnapshot();
  });
});
