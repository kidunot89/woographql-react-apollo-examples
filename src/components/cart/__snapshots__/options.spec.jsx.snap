// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Options should render successfully 1`] = `
<body>
  <div>
    <form
      class="w-full
    flex
    flex-wrap 
    gap-y-5 
    items-center 
    justify-between
    order-5
    md:justify-start
    md:mt-auto
    undefined
    "
    >
      <div
        class="w-full"
      >
        <div
          class="flex flex-wrap justify-start gap-x-5 mb-5"
        >
          <div
            class="text-lg font-bold italic w-full flex-none"
          />
          <div
            class="radio-button-wrapper shrink"
          >
            <label
              class="radio-button-label"
            >
              <input
                checked=""
                class="hidden"
                name="Print"
                type="radio"
                value="ankara"
              />
              <span
                class="radio-button"
              />
              Ankara
            </label>
          </div>
          <div
            class="radio-button-wrapper shrink"
          >
            <label
              class="radio-button-label"
            >
              <input
                class="hidden"
                name="Print"
                type="radio"
                value="candy-skull"
              />
              <span
                class="radio-button"
              />
              Candy skull
            </label>
          </div>
        </div>
      </div>
      <span
        class=" inline-block font-sans font-semibold text-xl text-center w-2/12"
      >
        $
        10
      </span>
      <div
        class="inline-flex gap-x-3 items-center w-9/12 ml-auto"
      >
        <span
          class="font-sans font-bold"
        >
          1
        </span>
        <div
          class="relative inline-block w-20"
        >
          <div
            class="appearance-none block text-center leading-5 m-0 py-2.5 px-0 h-8 outline-transparent outline outline-2 outline-offset-2 border rounded quantity-input"
            readonly=""
            type="number"
            value="1"
          />
          <div
            class="icon-button absolute z-10 left-2.5 top-1/2 -translate-y-1/2"
            role="button"
            tabindex="0"
          >
            <i
              class="fa-solid fa-minus"
            />
          </div>
          <div
            class="icon-button absolute z-10 right-2.5 top-1/2 -translate-y-1/2"
            role="button"
            tabindex="0"
          >
            <i
              class="fa-solid fa-plus"
            />
          </div>
        </div>
      </div>
      <button
        class="primary-button
        disabled:bg-gray-500
        w-full"
        type="submit"
      >
        Add To Basket
      </button>
    </form>
  </div>
</body>
`;
